### PR TITLE
Complete Homework#6

### PR DESCRIPTION
**Pull request by AlexGnatuyk**

### CODE REVIEW

File: Homework-7/Homework-7/Customer.cs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        public override string ToString()
+        {
+            return this.ToString("ALL", null);
+        }
+
+        public string ToString(string format)
+        {
+            return this.ToString(format, null);
+        }
+        public string ToString(string format, IFormatProvider formatProvider)
```
Comment (NikolayTur) [inline Homework-7/Homework-7/Customer.cs]: It's not a Strategy pattern. If I want to get some new format, I have to change Customer class.

File: Homework-7/Homework-7/Customer.cs [hunk 2]
```diff
@@ -52,38 +47,23 @@ public string ToString(string format, IFormatProvider formatProvider)
             {
                 case "ALL":
                 {
-                    string result = "";
-                    if (this.Name != null) result = String.Concat(result,this.Name)+" ";
-                    if (this.ContactPhone != null) result = String.Concat(result,this.ContactPhone)+" ";
-                    result = String.Concat(result, this.Revenue);
-                    
-                    return result;
+                   return "Customer record:" + Name + " " + ContactPhone + " " + Revenue.ToString("C");
```
Comment (NikolayTur) [inline Homework-7/Homework-7/Customer.cs]: Use string,Format()

File: Homework-7/Homework-7/Customer.cs [hunk 3]
```diff
@@ -52,38 +47,23 @@ public string ToString(string format, IFormatProvider formatProvider)
...
-                    string result = "";
-                    if (this.Name != null) result = String.Concat(result, this.Name);
-                    
-                    return result;
+                   return "Customer record:" + Name;
                 }
                 case "REVENUE":
                 {
-                    return this.Revenue.ToString(CultureInfo.CurrentCulture);
+                    return "Customer record:" + String.Format("C",Revenue);
```
Comment (NikolayTur) [inline Homework-7/Homework-7/Customer.cs]: Move `"Customer record:"` to `String.Format("C",Revenue)`

File: Homework-7/Homework-7/Customer.cs [hunk 4]
```diff
@@ -47,23 +47,23 @@ public string ToString(string format, IFormatProvider formatProvider)
             {
                 case "ALL":
                 {
-                   return "Customer record:" + Name + " " + ContactPhone + " " + Revenue.ToString("C");
+                   return "Customer record: " + Name + " " + ContactPhone + " " + Revenue.ToString("C");
```
Comment (NikolayTur) [inline Homework-7/Homework-7/Customer.cs]: Use string.Format()

File: Homework-6/Homework-6-2/Circle.cs [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Homework_6_2
+{
+    class Circle : Shape
+    {
+        public int R { get; set;}
+
+        public double Area()
```
[the remark this answers is not in the document]
Comment (AlexGnatuyk) [reply Homework-6/Homework-6-2/Circle.cs]: You, probably did not see my last commit of homework 6. 

https://github.com/net-fundamentals-2017/homework/blob/Aleksandr-Gnatyuk-Dev/Homework-6/Homework-6-2/Circle.cs

File: Homework-6/Homework-6-2/Circle.cs [hunk 6]
```diff
@@ -6,9 +6,18 @@
...
     {
-        public int R { get; set;}
+        private int r;
+
+        public int R
+        {
+            get => r;
+            set
+            {
+                if (value > 0) r = value;
```
[the remark this answers is not in the document]
Comment (AlexGnatuyk) [reply Homework-6/Homework-6-2/Circle.cs]: Last commit in homework 6 already covered this case

File: Homework-6/Homework-6-2/Circle.cs [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    class Circle : Shape
+    {
+        public int R { get; set;}
+
+        public double Area()
+        {
+            return 3.14 * R * R;
+        }
+
+        public double Perimetr()
```
[the remark this answers is not in the document]
Comment (AlexGnatuyk) [reply Homework-6/Homework-6-2/Circle.cs]: Last commit in homework 6 already covered this

File: Homework-6/Homework-6-2/Square.cs [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+{
+    class Square : Rectangle
+    {
+        public double Area()
+        {
+            if (X == Y)
+            {
+                return X * Y;
+            }
+            else return 0;
```
[the remark this answers is not in the document]
Comment (AlexGnatuyk) [reply Homework-6/Homework-6-2/Square.cs]: Last commit in homework 6 already covered this case. There I'm use

    'public Square(double height) : base(height, height)
            {
            }
     '

**Pull request merged**